plugins/modules/ndo_l3out_template.py

```diff
@@ -0,0 +1,936 @@
+                    payload["exportRouteMapRef"] = l3out_template_object.get_route_map(
+                        "outbound_route_map",
+                        tenant_id,
+                        tenant_name,
+                        bgp.get("outbound_route_map"),
+                        route_map_objects,
+                    ).get("uuid", "")
+
+                if bgp.get("route_dampening_ipv4"):
+                    payload["advancedRouteMapRefs"]["routeDampeningV4Ref"] = l3out_template_object.get_route_map(
+                        "route_dampening_ipv4",
+                        tenant_id,
+                        tenant_name,
+                        bgp.get("route_dampening_ipv4"),
+                        route_map_objects,
+                    ).get("uuid", "")
+
+                if bgp.get("route_dampening_ipv6"):
+                    payload["advancedRouteMapRefs"]["routeDampeningV6Ref"] = l3out_template_object.get_route_map(
+                        "route_dampening_ipv6",
+                        tenant_id,
+                        tenant_name,
+                        bgp.get("route_dampening_ipv6"),
+                        route_map_objects,
+                    ).get("uuid", "")
+
+            if ospf:
+                if not (ospf.get("cost") or ospf.get("area_id") or ospf.get("area_type")):
+                    mso.fail_json(msg="The O(ospf) - area_id, area_type, cost is required during the creation.")
+
+                payload["ospfAreaConfig"] = dict(
+                    cost=ospf.get("cost"),
+                    id=ospf.get("area_id"),
+                    areaType=ospf.get("area_type"),
+                )
+
+                default_route_leak = dict()
+                if ospf.get("originate_default_route"):
+                    default_route_leak["originateDefaultRoute"] = ORIGINATE_DEFAULT_ROUTE.get(ospf.get("originate_default_route"))
+
+                if ospf.get("originate_default_route_always") is not None:
+                    default_route_leak["always"] = ospf.get("originate_default_route_always")
+
+                if default_route_leak:
+                    payload["defaultRouteLeak"] = default_route_leak
+
+                redistribute = ospf.get("send_redistributed_lsas")
+                originate = ospf.get("originate_summary_lsa")
+                suppress_fa = ospf.get("suppress_forwarding_addr_translated_lsa")
+
+                control = dict()
+                if redistribute is not None:
+                    control["redistribute"] = redistribute
+
+                if originate is not None:
+                    control["originate"] = originate
+
+                if suppress_fa is not None:
+                    control["suppressFA"] = suppress_fa
+
+                if control:
+                    payload["ospfAreaConfig"]["control"] = control
+
+            mso.sanitize(payload)
+            ops = [dict(op="add", path="/l3outTemplate/l3outs/-", value=payload)]
```

Comment:
why not append to the already existing ops? as declared just before the if state == "present"